lib/foreman/renderer.rb

```diff
@@ -129,6 +129,7 @@ def load_template_vars
       # and is needed to direct the host to the correct url. without it, we increase
       # latency by requesting the correct url directly from the proxy.
       @template_url = params['url']
+      pxe_config
```

Comment:
Er, nearly, you'll see when you look.  The load_template_vars method is only called on certain actions, but could now be called on both PROVISION and CONFIG_URLS.